src/mscorlib/shared/System/ReadOnlySpan.cs

```diff
@@ -251,9 +251,21 @@ public override int GetHashCode()
         }
 
         /// <summary>
-        /// Returns a <see cref="String"/> with the name of the type and the number of elements
+        /// For <see cref="ReadOnlySpan{Char}"/>, returns a new instance of string that represents the characters pointed to by the span.
+        /// Otherwise, returns a <see cref="String"/> with the name of the type and the number of elements.
         /// </summary>
-        public override string ToString() => string.Format("System.ReadOnlySpan<{0}>[{1}]", typeof(T).Name, _length);
+        public override string ToString()
+        {
+            if (typeof(T) == typeof(char))
+            {
+                unsafe
+                {
+                    fixed (char* src = &Unsafe.As<T, char>(ref _pointer.Value))
+                        return new string(src, 0, _length);
+                }
+            }
+            return string.Format("System.ReadOnlySpan<{0}>[{1}]", typeof(T).Name, _length);
```

Comment:
Nit: I suggest changing `_length` to `_length.ToString()`.  Allocations here aren't a huge deal, but we might as well avoid the one produced by boxing `_length` into an object.  (Same nit for the span version below.)

Reply:
Although, actually, I just realized I've not revisited that notion since @justinvp's changes around using the new TryFormat overloads in the StringBuilder / string.Format implementation.  It's possible the box is now a cheaper approach.

Reply:
Currently, it will still box. Haven't gotten around to that optimization yet.

Reply:
> Currently, it will still box. Haven't gotten around to that optimization yet.

That's not actually what I meant.  If we call ToString, we allocate a string of length large enough to store the integer.  If we don't call ToString, it'll box, but then with your optimization it likely won't result in a ToString call, right?  And rather it'll just format into the StringBuilder it already has?  If so, at that point we're weighing the box vs the string.

Reply:
> That's not actually what I meant.

Ah, ok.

> If we don't call ToString, it'll box, but then with your optimization it likely won't result in a ToString call, right? And rather it'll just format into the StringBuilder it already has? If so, at that point we're weighing the box vs the string.

Right.

Reply:
Ok, so @ahsonkhan, ignore my original comment.  It's probably better the way it is :)  Even a one character string is a larger allocation than a box for an int.

Reply:
Sounds good.